containers/EditFly.js

```diff
@@ -1,10 +1,78 @@
-import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import React, { Component }  from 'react';
+import FlyForm from '../components/FlyForm';
+import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
+import { setCurrentFly } from '../actions'
+import { addUpdatedFly } from "../ApiCalls";
+import { connect } from "react-redux";
 
-export default function EditFly({ navigation }) {
-    return (
-        <View>
-            <Text>Edit Fly</Text>
+class EditFly extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            selectedFly: null
+        }
+    }
+
+    componentDidMount = () => {
+        const selectedFly = this.props.currentFlies.find(fly => fly.id === event.target.id)
+        this.setState({selectedFly: selectedFly})
+        this.setCurrentFly(selectedFly)
+    }
+
+    setCurrentFly = (selectedFly) => {
+        this.props.setCurrentFly(selectedFly.attributes)
+    }
+
+    updateFly = () => {
+        addUpdatedFly(this.state.selectedFly)
+    }
+
+    render() {
+       return (
+        <View style={styles.container}>
+            <FlyForm />
+            <View style={{flex: 1, flexDirection: 'row', justifyContent: 'center'}}>
+                <TouchableOpacity style={styles.button}
+                    onPress = {() => {
+                    this.props.navigation.navigate('MyFlyBox');
+                    }}>
```

Comment:
For this onPess section, we may need to use the clearFields when leaving the Edit Fly Page so that the currentFlyEntry in the Redux store is clear.